cmd/infra/powervs/destroy.go

```diff
@@ -371,8 +376,64 @@ func deleteSecrets(name, namespace, cloudInstanceID string, accountID string, re
 	return nil
 }
 
+// destroyPowerVsDhcpServer destroying powervs dhcp server
+func destroyPowerVsDhcpServer(ctx context.Context, logger logr.Logger, cloudInstanceID string, session *ibmpisession.IBMPISession) error {
```

Comment:
You need to consider the existing resource use case here. So you can't delete the resource always.

Reply:
Added the check!
Btw, the thought process behind not considering it was, shouldn't `skipPowerVs ` be set to `true` in case when the `powerVsCloudInstanceID` is user provided. So, if it is true, these lines should not get called and so, I left it.